Clarify remote-storage concerns (#6539)
elastic/docs-content: deploy-manage/production-guidance/optimize-performance/indexing-speed.md

  ece: all
```

> {{es}} clusters using directly-attached (local) storage generally perform better than those using remote storage. This is typically because direct storage achieves lower latency on I/O operations. Remote storage can often achieve very high throughput, but this is less important to most {{es}} workloads than the latency on individual operations.

Reviewer: Suggesting a rewrite for clarity and conciseness, flipping the focus on the direct storage benefit:

```suggestion
{{es}} clusters using directly-attached (local) storage generally perform better than those using remote storage. Direct storage typically provides lower latency for I/O operations, which is more critical for most {{es}} workloads than the high throughput that remote storage can often achieve.
```
